src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,158 @@
+        } else {
+            checkTail.next = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index >= 0 && index <= size) {
+            if (index == size) {
+                add(value);
+            } else {
+                Node<T> next = getIndexedNode(index);
+                Node<T> prev = next.prev;
+                Node<T> newNode = new Node<>(prev, value, next);
+                next.prev = newNode;
+                if (prev == null) {
+                    head = newNode;
+                } else {
+                    prev.next = newNode;
+                }
+                size++;
+            }
+        } else {
+            throw new IndexOutOfBoundsException("Can't add value. Pass correct input index");
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= 0 && index < size) {
+            return getIndexedNode(index).item;
+        }
+        throw new IndexOutOfBoundsException("Can't get item. Pass correct input index");
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (index >= 0 && index < size) {
+            Node<T> targetNode = getIndexedNode(index);
+            T oldValue = targetNode.item;
+            targetNode.item = value;
+            return oldValue;
+        }
+        throw new IndexOutOfBoundsException("Can't set value. Pass correct input index");
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (index >= 0 && index < size) {
+            return unlink(getIndexedNode(index));
+        }
+        throw new IndexOutOfBoundsException("Can't remove item. Pass correct input index");
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (Node<T> node = head; node != null; node = node.next) {
+                if (node.item == null) {
+                    unlink(node);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> node = head; node != null; node = node.next) {
+                if (object.equals(node.item)) {
+                    unlink(node);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
+    private T unlink(Node<T> node) {
+        T item = node.item;
+        Node<T> prev = node.prev;
+        Node<T> next = node.next;
+
+        if (prev == null) {
+            head = next;
+        } else {
+            prev.next = next;
+            node.prev = null;
+        }
+
+        if (next == null) {
+            tail = prev;
+        } else {
+            next.prev = prev;
+            node.next = null;
+        }
+
+        node.item = null;
+        size--;
+        return item;
+    }
```

Comment:
```suggestion
    private void unlink(Node<T> node) {
        Node<T> prev = node.prev;
        Node<T> next = node.next;
        if (prev == null) {
            head = next;
        } else {
            prev.next = next;
        }
        if (next == null) {
            tail = prev;
        } else {
            next.prev = prev;
        }
        size--;
    }
```
please remove redundant empty lines, also there is no need to remove links, that are pointing FROM the node we want to remove, it's enough to remove all links that point TO the node we want to remove
if some object (in our case Node) doesn't have a reference it will be automatically removed by garbage collector
why do you return item from this methods? maybe it's better to make this method void?